website/docs/cookbooks/introduction/formatting.md

```diff
@@ -0,0 +1,178 @@
+---
+title: Formatting
+sidebar_position: 5
+---
+
+import {ChainedSnippets} from "../../../src/components/MarkdownComponents.js";
+
+Scala CLI also lets you format your code using the [Scalafmt](https://scalameta.org/scalafmt/) formatter.
+
+
+## Basic usage
+
+This section describes how to use Scala CLI to format your code using default rules. For advanced usage and all available flags, please see next section.
+
+### Check
+
+To check the format of all your sources in the current directory:
+```bash
+scala-cli fmt --check .
+```
+
+This command will check if the code is formatted correctly according to the default rules.
+If it is not, it will return a diff of the changes that need to be made to match formatting rules.
+
+### Format
+
+If you want to format all your code, you can use the following command:
+```bash
+scala-cli fmt .
+```
+This command will format all your sources in the current directory according to the default rules.
+
+### Example
+
+Let's take a simple `HelloWorld` application:
+```scala title=HelloWorld.scala
+// use scala 3.3.0
+@main def hello()={println("Hello, World!")}
+def someLongMethodMoreThan80Characters():Unit={println("This is a long method")
+println("Another print here?")}
+```
+
+You can notice that this implementations is written with long lines, no indentation, no spaces and no new lines.
+It is hard to distinguish where one statement ends and another begins, which makes it hard to read.
+
+<ChainedSnippets>
+
+Let's check the format of this code:
+```bash fail
+scala-cli fmt --check .
+```
+Here is the output:
+```diff
+--- a/private/tmp/formatting-example/FormattingExample.scala
++++ b/private/tmp/formatting-example/FormattingExample.scala
+@@ -1,4 +1,7 @@
+ // use scala 3.3.0
+-@main def hello()={println("Hello, World!")}
+-def someLongMethodMoreThan80Characters():Unit={println("This is a long method")
+-println("Another print here?")}
++@main def hello() = { println("Hello, World!") }
++def someLongMethodMoreThan80Characters(): Unit = {
++  println("This is a long method")
++  println("Another print here?")
++}
++
+error: --test failed
+```
+</ChainedSnippets>
+
+As we can see, the code is not correctly formatted according to the default rules and [Scalafmt](https://scalameta.org/scalafmt/) formatter provides a diff of the changes that need to be made to match them.
+
+<ChainedSnippets>
+To apply the suggested changes to our code, we can run:
+
+```bash
+scala-cli fmt .
+```
```

Comment:
The command:
```bash
mill -i 'docs-tests[]'.test 'sclicheck.DocTests.cookbook introduction/formatting'
```
passes locally, I will push, and if the CI is red again, I will look into it deeper.

Reply:
yeah, this is weird.
I ran it locally myself and it's passing for me.
I pushed a commit ignoring the failing snippet, let's see how the CI behaves. 
I'll try to help you to get it to pass.

Reply:
right. I believe the `.` is catching some extra files on the CI, which makes it fail.
I narrowed it down to the example file, I'm thinking this should do the trick.

Reply:
Yes, that is a good assumption, thank you! I have read a lot of logs and [here](https://github.com/VirtusLab/scala-cli/actions/runs/9094838686/job/24996765911?pr=2903#step:6:2244) it can see no file at all. And I don't understand fully why 😅 . Also, the narrowing leads to the change in the examples which some may find unintuitive. 

We can mark this failing bash command on the 76 lines with the `fail` keyword and, I think it will pass the test with old examples. However, this will break local tests, but work in CI. I will try to reset scopes or modify the name of the file to pass the tests.

Reply:
I've adjusted the file's name and scopes, and it appears to be functioning properly now.